LiteDB/Engine/LiteEngine.cs

```diff
@@ -81,9 +79,14 @@ public LiteEngine(EngineSettings settings)
 
         #region Open & Close
 
+        public bool IsDisposed => _state.Disposed;
+
+        private readonly TaskCompletionSource<bool> _closedTask = new TaskCompletionSource<bool>();
+        public Task<bool> Closed => _closedTask.Task; 
```

Comment:
give a chance to outside observer to check 
if (db.IsDisposed)
{
  await db.Closed;
}
recreate db